### PR TITLE
Progress bar for downloads

### PR DESCRIPTION
**Pull request by Mandera**
After opening #388 I thought I could try doing it myself, this is the result!

### The only modified part is `WDMHttpClient.get`

- Added [tqdm](https://github.com/tqdm/tqdm) as dependency
- Changed `requests.get`'s `stream` parameter to True
  - I saw that this was the case some versions ago, hopefully, it's alright to go back to having it enabled again
- The progress bar is only shown if `Content-Length` exists in the response header and if it's greater than 100
  - Didn't want the progress bar to be shown for the "latest driver version" request which was very small
  - The value 100 should probably not be hard-coded
  - 100 might also not be the ideal default value, I only tried it on the Chrome driver
- Download speed unit is dynamic, I used the parameters from the [bytes CLI options](https://github.com/tqdm/tqdm#extra-cli-options) 
- In my attempt to mess as little as possible with the other code I'm storing the content from the stream to a bytearray
  - This bytearray then replaces the protected attribute `_content` in the response object
  - [Looks pretty safe to do](https://github.com/psf/requests/blob/da9996fe4dc63356e9467d0a5e10df3d89a8528e/requests/models.py#L889)
- I'm not too familiar with logging so not sure how to make the progress bar go through that if necessary
  - Found [this](https://stackoverflow.com/a/38739634/3936044) on SO anyhow
- Not sure how to make a unit test for this

Here's an example of how it looks
progress

Thank you

### CODE REVIEW

File: webdriver_manager/core/http.py [hunk 1]
```diff
@@ -29,6 +31,18 @@ def __init__(self):
         self._ssl_verify = ssl_verify()
 
     def get(self, url, **kwargs) -> Response:
-        resp = requests.get(url=url, verify=self._ssl_verify, **kwargs)
+        resp = requests.get(url=url, verify=self._ssl_verify, stream=True, **kwargs)
         self.validate_response(resp)
+
+        total = int(resp.headers.get("Content-Length", 0))
```
Comment (SergeyPirogov) [inline webdriver_manager/core/http.py:37]: can we move this to a method like show_progress or something like that?

**Review by SergeyPirogov — APPROVED** [review]

File: webdriver_manager/core/http.py (hunk 1 above)
Comment (Mandera) [reply webdriver_manager/core/http.py]: Sure thing, I added it in utils.py though since it was static anyway

**Comment by Mandera** [comment]
- Moved progress bar code to show_download_progress() in utils.py
- Hardcoded the "[WDM] - " prefix for the progress bar as I think making tqdm use the logger's formatting dynamically isn't feasible
- Added tqdm to install_requires in setup.py (In addition to Pipfile) - Is this the right way to do it? Seems strange to define deps in two places

**Comment by SergeyPirogov** [comment]
Well, I think that this approach shold be just documented. I dont want to add 3rd party dependency and have a progress bar enabled by default

**Comment by Mandera** [comment]
Yeah makes sense to have it disabled by default, add a parameter `progress_bar=False´ to `DriverManager` and its subclasses?

Are you open to making tqdm an optional dependency? Such as `pip install webdriver-manager[progress-bar]` or `pip install webdriver-manager[full]`
If not I could try writing a simplistic progress bar without 3rd party deps

**Comment by SergeyPirogov** [comment]
> Yeah makes sense to have it disabled by default, add a parameter `progress_bar=False` to `DriverManager` and its subclasses?
> 
> Are you open to making tqdm an optional dependency? Such as `pip install webdriver-manager[progress-bar]` or `pip install webdriver-manager[full]` If not I could try writing a simplistic progress bar without 3rd party deps

Yes, making it optional is a great idea

**Pull request merged**